### PR TITLE
Cc 7146/convert consul hcp to a simpler component for some upcoming changes

### PR DESCRIPTION
**Pull request by chris-hut**
### Description
As part of [CC-7146](https://hashicorp.atlassian.net/browse/CC-7146) we'll be adding a sidebar item about the cluster's link status and need to modify the logic from the existing `Back to Consul` Nav bar item.

This PR refactors the component that renders the `Back to consul` item in the following ways:
- Moves it to the `consul-ui` package, for easier use and editing
- Kills the now empty `consul-hcp` package to clean up a little bit

### Testing & Reproduction steps
- `yarn start`
- In your browser's dev tools run the following to set/unset the following ENV variables:
  - To enable: `Scenario('CONSUL_HCP_ENABLE=1');` and `Scenario('CONSUL_HCP_URL=https://corn.com')` 
  - To disable: delete them from your cookies
- When enabled, you should see the classic `<- Back to HCP` nav bar item
- When disabled, that will be hidden

image

### Links
- [Jira](https://hashicorp.atlassian.net/browse/CC-7146)

### PR Checklist

* [x] updated test coverage

### References
image

[CC-7146]: https://hashicorp.atlassian.net/browse/CC-7146?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by WenInCode** [comment]
@hut just taking a look now. 

Is there a reason this is backported to 1.17 and not just applied to the new 1.18 release? I haven't followed past PRs so apologies if this works was always planned to be backported to 1.17.x

File: ui/packages/consul-ui/app/components/hcp-nav-item/index.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Component from '@glimmer/component';
```
Comment (WenInCode) [inline ui/packages/consul-ui/app/components/hcp-nav-item/index.js:1]: I think this file is missing the copyright header

**Review by WenInCode — APPROVED** [review]
LGTM, question regarding license and backporting aside.

**Review by WenInCode — COMMENTED** [review]
I think you may want to also remove this bit? 

https://github.com/hashicorp/consul/blob/f42152a6e52efe0dfe9b2da6a0f4d3d85854f799/ui/packages/consul-ui/lib/startup/templates/body.html.js#L111-L114

Honestly, confusing. It looks like a IIFE that is called with an object of the features/packages that should be checked if they need to be included but we are also manually doing that above? I'm thinking it may be something that runs for either configuration enabled or if they are enabled by env vars? Not sure right now.

**Review by WenInCode — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@chris-hut, a backport is missing for this PR [20344] for versions [1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Pull request merged**